File: app/src/main/java/com/scottmangiapane/cardclicks/CardViewModel.java
package com.scottmangiapane.cardclicks;

import androidx.lifecycle.ViewModel;

import com.scottmangiapane.cardclicks.game.Game;

public class CardViewModel extends ViewModel {
    private final int defaultSize;
    private final long defaultTimeRemaining;

    public Game game;
    public boolean isGameRunning;
    public int score;
    public long timeRemaining;

    public CardViewModel(int size, long timeRemaining) {
        this.defaultSize = size;
        this.defaultTimeRemaining = timeRemaining;
        reset();
    }

    public void reset() {
        this.game = new Game(defaultSize);
        this.isGameRunning = true;
        this.score = 0;
        this.timeRemaining = defaultTimeRemaining;
    }
}

File: app/src/main/java/com/scottmangiapane/cardclicks/CardViewModel.kt
package com.scottmangiapane.cardclicks

import androidx.lifecycle.ViewModel
import com.scottmangiapane.cardclicks.game.Game

class CardViewModel(private val defaultSize: Int, private val defaultTimeRemaining: Long) :
    ViewModel() {
    lateinit var game: Game
    var isGameRunning = false
    var score = 0
    var timeRemaining: Long = 0

    init {
        reset()
    }

    fun reset() {
        game = Game(defaultSize)
        isGameRunning = true
        score = 0
        timeRemaining = defaultTimeRemaining
    }
}
